nsot/api/views.py

```diff
@@ -495,32 +512,45 @@ def supernets(self, request, pk=None, site_pk=None, *args, **kwargs):
 
         return self.list(request, queryset=networks, *args, **kwargs)
 
-    @detail_route(methods=['get'])
+    @detail_route(methods=['get', 'post'])
     def next_network(self, request, pk=None, site_pk=None, *args, **kwargs):
         """Return next available networks from this Network."""
         network = self.get_resource_object(pk, site_pk)
-
         params = request.query_params
         prefix_length = params.get('prefix_length')
         num = params.get('num')
         strict = qpbool(params.get('strict_allocation', False))
         networks = network.get_next_network(
-            prefix_length, num, strict, as_objects=False
+            prefix_length, num, strict, as_objects=True
         )
-
-        return self.success(networks)
-
-    @detail_route(methods=['get'])
+        if request.method == 'POST':
+            if qpbool(params.get('reserve', False)):
+                state = models.Network.RESERVED
+            else:
+                state = models.Network.ALLOCATED
+            self.allocate_networks(networks, prefix_length, site_pk, network, state=state)
+        return self.success([unicode(x) for x in networks])
+
+    @detail_route(methods=['get', 'post'])
     def next_address(self, request, pk=None, site_pk=None, *args, **kwargs):
         """Return next available IPs from this Network."""
         network = self.get_resource_object(pk, site_pk)
-
         params = request.query_params
         num = params.get('num')
         strict = qpbool(params.get('strict_allocation', False))
-        addresses = network.get_next_address(num, strict, as_objects=False)
-
-        return self.success(addresses)
+        addresses = network.get_next_address(num, strict, as_objects=True)
```

Comment:
If we ditch `prefix_length` as an argument to `.allocate_networks()` we can probably go back to returning a list of unicode strings (`as_objects=False`) and then the version/prefix_length calculation can also go away!